### PR TITLE
crimson/os/seastore/btree: don't update internal mapping with invalid iterators

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Signed-off-by: Xuehan Xu <xxhdx1985126@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/btree/fixed_kv_btree.h [hunk 1]
```diff
@@ -968,7 +968,7 @@ class FixedKVBtree {
           parent.node
         );
         typename internal_node_t::Ref mparent = mut->cast<internal_node_t>();
-        mparent->update(piter, new_addr);
+        mparent->update(mparent->iter_idx(piter.offset), new_addr);
```
Comment (xxhdx1985126) [inline src/crimson/os/seastore/btree/fixed_kv_btree.h:971]: piter may have been invalid at this point, but it's very strange that it hasn't caused any failure, @athanatos @cyx1231st do you guys have any clue about why?

**Comment by xxhdx1985126** [comment]
jenkins test make check

File: src/crimson/os/seastore/btree/fixed_kv_btree.h (hunk 1 above)
Comment (cyx1231st) [reply src/crimson/os/seastore/btree/fixed_kv_btree.h:971]: The parent should have been added to the transaction's read-set, right? Why is it possible for the transaction to see an invalidated extent when it is in the read-set? `si_then()` should be able to interrupt the invalidated transaction's future chain before that.

This modification seems identical to the original version, refer to https://github.com/ceph/ceph/blob/f7e122e4b8f671e0555855c0a53d9f748a35dcfb/src/crimson/common/fixed_kv_node_layout.h#L313-L317

Comment (xxhdx1985126) [reply src/crimson/os/seastore/btree/fixed_kv_btree.h:971]: Ah, yes, I thought `FixedKVInternalNode::update` was `FixedKVNodeLayout::update`. I'm closing this pr.

**Pull request closed without merging**